# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 8b04d03ac19e
# Full Hash: 8b04d03ac19eb4fbeb37de4dde03d30b1f74868a
# Author: stransky <stransky@redhat.com>
# Date: 2022-02-02 04:09:16
# Description:
#   Bug 1747475 [Linux] Don't assert/crash if we're missing mGdkWindow r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137554
# ==============================================================================

diff -r 2c40cadf59ff -r 8b04d03ac19e widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Tue Feb 01 18:59:05 2022 +0000
+++ b/widget/gtk/nsWindow.cpp	Tue Feb 01 19:12:32 2022 +0000
@@ -5253,7 +5253,8 @@
 }
 
 Window nsWindow::GetX11Window() {
-  return GdkIsX11Display() ? gdk_x11_window_get_xid(mGdkWindow) : X11None;
+  return GdkIsX11Display() && mGdkWindow ? gdk_x11_window_get_xid(mGdkWindow)
+                                         : X11None;
 }
 
 void nsWindow::EnsureGdkWindow() {
@@ -5262,7 +5263,6 @@
                                                         : mShell);
     g_object_set_data(G_OBJECT(mGdkWindow), "nsWindow", this);
   }
-  MOZ_DIAGNOSTIC_ASSERT(mGdkWindow, "We're missing GdkWindow!");
 }
 
 bool nsWindow::GetShapedState() {
